app/services/lists/protocols_service.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Lists
+  class ProtocolsService < BaseService
+    PREFIXED_ID_SQL = "('#{Protocol::ID_PREFIX}' || COALESCE(\"protocols\".\"parent_id\", \"protocols\".\"id\"))".freeze
```

Comment:
Doesn't the Protocol model already have`PREFIXED_ID_SQL` defined?

Reply:
this to make proper search by id.
Protocol::ID_PREFIX - just 'PR' and then if parent_id exist it will attach parent id

Reply:
i used it from previous implementation